components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/executers/RemoteFetchTaskExecutor.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.remotefetch.common.RemoteFetchConfiguration;
+import org.wso2.carbon.identity.remotefetch.common.exceptions.RemoteFetchCoreException;
+import org.wso2.carbon.identity.remotefetch.core.executers.tasks.RemoteFetchConfigurationBatchTask;
+import org.wso2.carbon.identity.remotefetch.core.executers.tasks.RemoteFetchConfigurationImmediateTask;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The execution engine for Fetching the artifacts from remote repository and loads it to the system.
+ * This execution engine has functions to start batch task and immediate task.
+ * This execution engine starts batch task at component activation and schedule at fixed rate.
+ * {@link RemoteFetchConfigurationBatchTask}
+ * This execution engine starts immediate task when OSGi trigger service called.
+ * {@link RemoteFetchConfigurationImmediateTask}
+ *
+ */
+public class RemoteFetchTaskExecutor {
+
+    private static final Log log = LogFactory.getLog(RemoteFetchTaskExecutor.class);
+
+    private ScheduledExecutorService scheduler;
+    private RemoteFetchConfigurationBatchTask remoteFetchConfigurationBatchTask;
+
+    public RemoteFetchTaskExecutor() {
+
+    }
+
+    public void createScheduler() {
+
+        scheduler = Executors.newScheduledThreadPool(1);
+    }
+
+    /**
+     * Shutdown scheduler while deactivating the component.
+     */
+    public void shutdownScheduler() {
+
+        scheduler.shutdown();
+    }
+
+    /**
+     * Schedule Batch execution after checking whether enabled or not.
+     */
+    public void startBatchTaskExecution() {
+
+        remoteFetchConfigurationBatchTask = new RemoteFetchConfigurationBatchTask();
+        try {
+            scheduler.scheduleAtFixedRate(remoteFetchConfigurationBatchTask, 0, (60 * 1), TimeUnit.SECONDS);
+            log.info("Batch Task is scheduled.");
+        } catch (Exception e) {
+            log.error("Error while scheduling batch task", e);
+        }
+    }
+
+    /**
+     * Schedule immediate task execution when OSGi trigger service called.
+     * @param remoteFetchConfiguration
+     * @throws RemoteFetchCoreException
+     */
+    public void startImmediateTaskExecution(RemoteFetchConfiguration remoteFetchConfiguration)
+            throws RemoteFetchCoreException {
+
+        RemoteFetchConfigurationImmediateTask remoteFetchConfigurationImmediateTask =
+                new RemoteFetchConfigurationImmediateTask(remoteFetchConfiguration);
+
+        try {
+            scheduler.schedule(remoteFetchConfigurationImmediateTask, 0, TimeUnit.SECONDS);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Immediate Task is scheduled for remote fetch configuration "
+                        + remoteFetchConfiguration.getRemoteFetchConfigurationId());
+            }
+        } catch (Exception e) {
```

Comment:
Is it possible to handle specific exceptions rather than generic Exception here?

Reply:
Yes, rather than handle the generic exceptions, We can throw from the lower level and handle from Service Implementations.
Thank you, This comment and your previous comments were fixed in
https://github.com/wso2-extensions/identity-fetch-remote/pull/40
https://github.com/wso2-extensions/identity-fetch-remote/pull/41
https://github.com/wso2-extensions/identity-fetch-remote/pull/42
https://github.com/wso2-extensions/identity-fetch-remote/pull/43
https://github.com/wso2-extensions/identity-fetch-remote/pull/44
https://github.com/wso2-extensions/identity-fetch-remote/pull/45.
Hence I resolve this conversation.